eventauth/eventauth.go

```diff
@@ -251,6 +229,73 @@ func redactEventAllowed(event Event, authEvents AuthEvents) error {
 	panic("Not implemented")
 }
 
+// defaultEventAllowed checks whether the event is allowed by the default
+// checks for events.
+// It returns an error if the event is not allowed or if there was a
+// problem loading the auth events needed.
 func defaultEventAllowed(event Event, authEvents AuthEvents) error {
-	panic("Not implemented")
+	var allower eventAllower
+	if err := allower.setup(authEvents, event.Sender); err != nil {
+		return err
+	}
+
+	return allower.commonChecks(event)
+}
+
+// An eventAllower has the information needed to authorise all events types
+// other than m.room.create, m.room.member and m.room.alias which are special.
+type eventAllower struct {
+	create      createContent
+	member      memberContent
+	powerLevels powerLevelContent
+}
+
+// setup loads the necesary information from the auth events.
+func (e *eventAllower) setup(authEvents AuthEvents, senderID string) error {
```

Comment:
> necesary

Also, in general I thought "in order to set up this struct you need to execute this arbitrary code" stuff is typically done via a `New()` call which sets up the struct and returns the struct possibly with an error. Any reason for deviating from that convention here? Is it ever valid to be using an `eventAllower` *without* calling `setup()`?